client/login/wp-login/index.jsx

```diff
@@ -104,12 +104,16 @@ class Login extends React.Component {
 				</a>;
 		}
 
-		const goBackLink = ! magicLoginView && <a
-			href="#"
-			key="back-link"
-			onClick={ this.goBack }>
-				<Gridicon icon="arrow-left" size={ 18 } /> { this.props.translate( 'Return' ) }
-			</a>;
+		let goBackLink;
+		if ( typeof window === 'undefined' || window.history.length > 1 ) {
```

Comment:
`/login` is rendered on the server side, so `window` isn't available. However, when it gets to the client side, `window` is now available, and we can detect if there's something in `window.history` to go back to.

Unfortunately, this is both bad (it causes a the Return button to initially appear, then disappear) and wrong (React dumps an error in the console, complaining about markup that doesn't match between server and client).